### ISSUE TITLE
Java rules don't appear to be usable in OSS edition of buck2

### ISSUE THREAD

**Issue by zjturner** [facebook/buck2#627]
I tried to use `java_binary()`.  I was pretty quickly hit with an error about one of my provider fields (`JavaToolchainInfo.fat_jar`) being unset, so I looked into what it might be expecting, and found this.

https://github.com/facebook/buck2/blob/d458e16b57cf34853ff4b239cd609eabe235f257/prelude/java/java_binary.bzl#L40-L48

Whatever tool this is doesn't appear to be a standard JRE tool, and I suspect it's a Meta internal tool.  Which means all of the Java rules don't work outside of Meta.  Is my hunch correct about this, or is there a way to get this tool?

**Comment by IanChilds** [comment]
Yeah, that's correct. Java rules are not yet properly supported unfortunately

**Comment by zjturner** [comment]
That’s too bad.  Can you shed some light on why?  What kind of use cases does meta have for which the jre alone isn’t sufficient to build everything you need to build?

**Comment by navkast** [comment]
Just want to add my name to the list of people hoping for this. I'm representing Addepar, which is a long-time user / contributor to the original buck codebase. Would love to begin converting to buck2 😄

**Comment by oleksii-danchuk** [comment]
Tried it this weekend. It is better now. But prelude java toolchain is very tightly coupled to Android and Kotlin, even if you're just building plain Java code...
